taca/illumina/HiSeq_Runs.py

```diff
@@ -92,138 +92,155 @@ def compute_undetermined(self):
         NoIndexLanes = [lane["Lane"] for lane in self.runParserObj.samplesheet.data if "NoIndex" in lane["index"]]
         if len(NoIndexLanes) == 0:
             return True # everything is fine I can proceed to QC
-        else:
-            NoIndex_Undetermiend = os.path.join(self.run_dir, "Demultiplexing_NoIndex")
-            if not os.path.exists(NoIndex_Undetermiend):
-                os.makedirs(NoIndex_Undetermiend)
-                #for these lanes I have no undetermiend as I demux them without index.
-                #now geenrate the base masks per lane
-                per_lane_base_masks = self._generate_per_lane_base_mask()
-                #store here only the NoIndex lanes
-                per_lane_base_masks_NoIndex = {}
-                for NoIndexLane in NoIndexLanes:
-                    per_lane_base_masks_NoIndex[NoIndexLane] = per_lane_base_masks[NoIndexLane]
-                    base_mask_key = per_lane_base_masks[NoIndexLane].keys()[0]
-                    new_base_mask = []
+        #otherwise proceed
+        
+        NoIndex_Undetermiend = os.path.join(self.run_dir, "Demultiplexing_NoIndex")
+        if not os.path.exists(NoIndex_Undetermiend):
+            #for these lanes I have no undetermiend as I demux them without index.
+            #now geenrate the base masks per lane
+            per_lane_base_masks = self._generate_per_lane_base_mask()
+            #store here only the NoIndex lanes
+            per_lane_base_masks_NoIndex = {}
+            run_with_no_index = False # use this flag to check that we are not in the C.Daub case
+            for NoIndexLane in NoIndexLanes:
+                per_lane_base_masks_NoIndex[NoIndexLane] = per_lane_base_masks[NoIndexLane]
+                base_mask_key = per_lane_base_masks[NoIndexLane].keys()[0]
+                new_base_mask = []
+                if len(per_lane_base_masks_NoIndex[NoIndexLane][base_mask_key]['base_mask']):
```

Comment:
I'm not sure I understand this if. if len(<something>) should be the same as if(something), because '' is evluated as false. Are you testing for a specific length ?